Role view improvements

SamLubbers opened the pull request:
Before making the task detail view (#7) it is a good idea to finalise the design for the role view. Then we can just copy the design for tasks. I made the following improvements to the role view:

1. Removed sidebar. There was too little content in the sidebar and too much content in the main text.
2. Added at the bottom of the description the time commitment information that used to be in the sidebar.
3. Added missing role fields (published date, requirements & responsibilities)
4. Added button to apply for the role. When this is clicked a modal shows with the contact details, each of which can be clicked. I acknowledge that having a modal inside of a modal is not an elegant solution, but it works fine for now. Also, I decided to add an "apply" button rather than directly adding the contact details for 2 reasons:
   4.1 I couldn't find a good place to add the contact details. Having the contact details on top didn't is too much because most people want to read the requirements/description first, having it on the bottom made it too hard to find and having them on the side like before created an imbalance of content (2.1)
   4.2 Another argument for having a button is that such a button will be needed when users apply for roles through the platform. If we have text now and later a button it can be confusing for users.
5. Added buttons for marking a role as taken, and buttons for edit and delete (first part of #19, #21 & #23). Note that these buttons will only show up for the accounts that create those roles. 

Lastly, I removed the nested CSS classes because they are not necessary in Vue components. The `scoped` attribute on the `style` tag ensures those classes are only available in that component. 

I would like to suggest that this PR closes #15, even though not all parts of that issue are completed. I will make separate individual issues for the parts that are not implemented yet.

JBuma commented:
I like how this looks, only thing i would change is move the `apply` and `role taken` buttons to a footer in the bottom, the other menu button fits okay there. People are more used to those action buttons being in the bottom right, so it would be more conventional that way. Also, if you use tab to select the buttons, they're in kind of a weird order (the apply buttons gets focus first, then the role taken button, then the menu button).

SamLubbers commented:
I added the buttons on top to make them easier to find but they indeed do not fit well there. Just added the buttons at the bottom and the tab issue is resolved.
